modules/storages/lib/api/v3/storages/storage_representer.rb

```diff
@@ -35,17 +35,46 @@ module API::V3::Storages
   URN_CONNECTION_AUTH_FAILED = "#{::API::V3::URN_PREFIX}storages:authorization:FailedAuthorization".freeze
   URN_CONNECTION_ERROR = "#{::API::V3::URN_PREFIX}storages:authorization:Error".freeze
 
+  URN_STORAGE_TYPE_NEXTCLOUD = "#{::API::V3::URN_PREFIX}storages:Nextcloud".freeze
+
+  STORAGE_TYPE_MAP = {
+    URN_STORAGE_TYPE_NEXTCLOUD => Storages::Storage::PROVIDER_TYPE_NEXTCLOUD
+  }.freeze
+
+  STORAGE_TYPE_URN_MAP = {
+    Storages::Storage::PROVIDER_TYPE_NEXTCLOUD => URN_STORAGE_TYPE_NEXTCLOUD
+  }.freeze
+
   class StorageRepresenter < ::API::Decorators::Single
     # LinkedResource module defines helper methods to describe attributes
     include API::Decorators::LinkedResource
     include API::Decorators::DateProperty
     include Storages::Peripherals::StorageUrlHelper
 
+    module ClassMethods
+      private
+
+      def link_without_resource(name, getter:, setter:)
+        link name do
+          instance_eval(&getter)
+        end
+
+        property name,
+                 exec_context: :decorator,
+                 getter: ->(*) {},
+                 setter:,
+                 skip_render: true,
+                 linked_resource: true
+      end
+    end
+
+    extend ClassMethods
+
     def initialize(model, current_user:, embed_links: nil)
       @connection_manager =
         ::OAuthClients::ConnectionManager.new(user: current_user, oauth_client: model.oauth_client)
 
-      super
+      super(model, current_user:, embed_links:)
```

Comment:
This change is unnecessary since calling `super` (without args) will just pass on all the args of the method `super` is called in.

Reply:
reverted